subgraph-js/src/printSubgraphSchema.ts

```diff
@@ -178,22 +178,30 @@ function printImplementedInterfaces(
     : '';
 }
 
-function printObject(type: GraphQLObjectType): string {
-  // Apollo change: print `extend` keyword on type extensions.
-  //
-  // The implementation assumes that an owned type will have fields defined
-  // since that is required for a valid schema. Types that are *only*
-  // extensions will not have fields on the astNode since that ast doesn't
-  // exist.
-  //
-  // XXX revist extension checking
-  const isExtension =
-    type.extensionASTNodes && type.astNode && !type.astNode.fields;
+/**
+ * Method that checks if a type is an extension
+ *
+ * The implementation assumes that an owned type will have fields defined
+ * since that is required for a valid schema. Types that are *only* extensions will:
+ *  - have the `extensionASTNodes` defined and with at least 1 field. This is because prior GraphQL v16
+ *    the `extensionASTNodes` property was undefined, where now it comes as an empty {@link Array}
+ *  - not have fields on the `astNode` since that Abstract Syntax Tree doesn't exist;
+ * @param type
+ * @return boolean
+ */
+function checksIfTypeIsExtension(type: GraphQLObjectType | GraphQLInterfaceType): boolean {
+  return (
+      Array.isArray(type.extensionASTNodes) &&
+      !!type.extensionASTNodes?.length &&
```

Comment:
```suggestion
      type.extensionASTNodes.length > 0 &&
```
Is this what we actually want?
* I'm not sure the optional chain is necessary since we've asserted on the line before that it's an `Array`
* I think `!!` will yield the same result in all cases, but `length > 0` seems a bit more semantically correct and expresses exactly what we're interested in.